[14.0][REF] include representative address type in cooperator

victor-champonnois opened the pull request:
fix https://github.com/coopiteasy/vertical-cooperative/issues/350
related to https://github.com/coopiteasy/addons/pull/262
12.0 version: #362

robinkeunen reviewed (approved):
LGTM 👌 can you add the note about the conflicting representative fields like in 12.0 ?

robinkeunen reviewed (changes requested):
LGTM 👌 can you add the note about the conflicting representative fields like in 12.0 ?

victor-champonnois commented:
> LGTM ok_hand can you add the note about the conflicting representative fields like in 12.0 ?

Done

robinkeunen reviewed (approved):
thx!

carmenbianca commented on cooperator/models/partner.py:
```diff
@@ -105,6 +105,10 @@ def _compute_share_info(self):
         store=True,
         compute="_compute_representative_of_member_company",
     )
+    type = fields.Selection(
+        selection_add=[("representative", "Representative")],
+        ondelete={"representative": "set null"},
+    )
```
I am missing a comment here that explains _why_ we need this type in `cooperator`.

victor-champonnois replied:
comment added

victor-champonnois replied:
The comment is not very precise, because it is not very clear if this field is indeed needed (see the discussion in the issue). The problem is also explained in the roadmap.

victor-champonnois replied:
also added in 12.0

carmenbianca replied:
@victor-champonnois I would do one of two things (or both):

1. Copy the explanation from the discussion verbatim into the comment.
2. Provide a link to the discussion in the comment.

Having this context would be really handy when/if we hit this problem again.

victor-champonnois replied:
added the link (the discussion is too long to be added in the code)

carmenbianca replied:
:thumbsup:

victor-champonnois committed:
[REF] include representative address type in cooperator

Put the representative type in cooperator module instead of in
partner_contact_address for inclusion in OCA.

huguesdk commented:
/ocabot merge minor
